### PR TITLE
Fixes and add method for downloading objects

### PR DESCRIPTION
**Pull request by monicatie**
Added some things i need for the blueprint forge job

- minor fixes to schema of the objects we get from forge
- Added functionality for downloading objects
- Added test for download

### CODE REVIEW

File: dm/common.go [hunk 1]
```diff
@@ -27,6 +27,13 @@ type DataDetails struct {
 	Links   	Link      	`json:"links, omitempty"`
 }
 
+type FolderContents struct {
```
Comment (monicatie) [inline dm/common.go:30]: Hub's DataDetails don't have an Included so I just made one specifically for FolderContents

File: dm/object.go [hunk 2]
```diff
@@ -136,3 +144,29 @@ func uploadObject(path, bucketKey, objectName string, dataContent io.Reader, tok
 	return
 
 }
+
+func downloadObject(path, bucketKey, objectName string, token string) (result io.ReadCloser, err error) {
+	task := http.Client{}
+
+	//dataContent := bytes.NewReader(data)
```
Comment (camillaol) [inline dm/object.go]: can/should we remove this comment?

Comment (monicatie) [reply dm/object.go]: ah yea

File: dm/object_test.go [hunk 3]
```diff
@@ -92,3 +92,66 @@ func TestBucketAPI_UploadObject(t *testing.T) {
...
+		}
+		defer deleteBucket(bucketAPI, tempBucket, t)
+	}
+
+	file, err := os.Open(testFilePath)
+	if err != nil {
+		t.Fatal("Cannot open testfile for reading")
+	}
+	defer file.Close()
+	// data, err := ioutil.ReadAll(file) // returns []byte
```
Comment (camillaol) [inline dm/object_test.go]: same here, can we dump this commented out line?

**Review by camillaol — APPROVED** [review]
LGTM. Per chat, we'll open issues for the failing bucket tests (which is from before our time)

**Pull request merged**